### PR TITLE
Complete Firebase Authentication including frontend and backend

### PR DESCRIPTION
**Pull request by 712Kunal**
### **Authentication using Firebase**

**Done with following things -** 

1. Updated UI of the login and the signup page. 
2. Navigating user from login to signup page and vice versa.
3. Added input validation using Regex and displaying it to the user.
4. Added error handling.
5. Added Firebase to do authentication including login and signup functionality.
6. Created a context to store the users data for the future use.

### **Screenshots of UI -**

**SIGNUP PAGE -**

image

**LOGIN PAGE -**

image

### CODE REVIEW

File: client/src/App.css
Comment (mendacium-a11y) [inline client/src/App.css:1]: pls remove this font import and use the ones that are being used throughout the project

**Review by mendacium-a11y — COMMENTED** [review]
Pls dont change the ui of the login page and use a simillar looking layout for the sign up page as well and add a video as well of signing up and using the same credentials to log in

**Comment by 712Kunal** [comment]
### Video of the authentication - 

https://drive.google.com/file/d/1bn4tW3OQrKYe_kA9lliBv1PGH9sPedfo/view?usp=drive_link

**Comment by mendacium-a11y** [comment]
@712Kunal 
- pls update the example.env as well
- once logged in, change the login button to log out and necessary functionality needed for it
- when on the sign up and login page, the footer shouldn't be showing sign up and login option respectively

**Comment by mendacium-a11y** [comment]
image
getting this error when i am trying to run the development server for the frontned

**Comment by 712Kunal** [comment]
The errors are comming from your local system only, because you might not have the packages whatever they are needed to run the app, follow the steps which are provided in the error message only and then check it 
image

First deleted the package-lock.json file and then deleted the package.json file and then run the commad npm install, so that all the required packages and dependencies will come in to your local system and then simply run the app.
I think it might work ....

**Comment by mendacium-a11y** [comment]
- I am using the package and package-lock files commtied by you

**Comment by 712Kunal** [comment]
First of all this is **not a development error** actually, its a **deployment error** because the app is **running successfully in my local system without any type of error .**

Check the node modules files generated in your machine from which you are testing, if it is not then it means the firebase package which I had installed during authentication will not come in your machine, using my package.json and package-lock.json is doesn't makes any sense that these packages will run in your local machine

To installed the packages or to get the node module of the project you have to run the command "**npm install**" in your terminal in the client folder only, so that all packages will get installed .

if still problem arises, **then only delete package-lock.json**  file (not package.json) and then run the "**npm install**" command .

And also let me know that how you are testing my code or reviewing it, I mean are you simply clonning my repo and then testing it in your local environment or merging my branch with your local main branch

**Comment by mendacium-a11y** [comment]
if it runs on your laptop and not mine, that means your code is not reproducible, I am running it on a github codespaces, so no scope of dependency issue from my side. You have an error in your `package-lock.json`, the one that **you** are commiting

Everything seems to be working fine now, that is the only issue that is holding me from merging your pr

**Comment by 712Kunal** [comment]
I had again regenerated the package-lock.json file, you can review it again

**Pull request merged**